ci/build/build-lib.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+# This is a library which contains functions used inside ci/build
+#
+# We separated it into it's own file so that we could easily unit test
+# these functions and helpers.
+
+# On some CPU architectures (notably node/uname "armv7l", default on Raspberry Pis),
+# different package managers have different labels for the same CPU (deb=armhf, rpm=armhfp).
+# This function returns the overriden arch on platforms
+# with alternate labels, or the same arch otherwise.
+get_nfpm_arch() {
+  case "$ARCH" in
```

Comment:
Alternatively you can pass the parameters as arguments, e.g. `get_nfpm_arch armv7l deb` and then use $1 and $2 instead of $ARCH and $PKG_FORMAT. This way you can call the function without modifying a global env var state. This is what this does: https://github.com/cdr/m/blob/f0de9bff47859bfe8965474b62a2cd9e900863d9/lib/bash/trace.sh#L23-L26

Reply:
OOO good idea. I'm all about not relying on global env vars. I'll refactor! Thanks for the suggestion!